packages/braid-design-system/src/lib/components/Page/Page.tsx

```diff
@@ -20,20 +21,31 @@ export const Page = ({
   footerPosition,
   data,
   ...restProps
-}: PageProps) => (
-  <Box
-    display="flex"
-    flexDirection="column"
-    className={styles.fullHeight}
-    {...buildDataAttributes({ data, validateRestProps: restProps })}
-  >
-    <Box
-      flexGrow={footerPosition !== 'belowFold' ? 1 : undefined}
-      className={footerPosition === 'belowFold' ? styles.fullHeight : undefined}
-      paddingBottom="xxlarge"
-    >
-      {children}
-    </Box>
-    {footer}
-  </Box>
-);
+}: PageProps) => {
+  assert(
+    !useContext(PageContext),
+    'Page components should not be nested within each other',
+  );
+
+  return (
+    <PageContext.Provider value={true}>
+      <Box
+        display="flex"
+        flexDirection="column"
+        className={styles.fullHeight}
+        {...buildDataAttributes({ data, validateRestProps: restProps })}
+      >
+        <Box
+          flexGrow={footerPosition !== 'belowFold' ? 1 : undefined}
+          className={
+            footerPosition === 'belowFold' ? styles.fullHeight : undefined
+          }
+          paddingBottom="xxlarge"
```

Comment:
Are consumers going to want to customize this padding?

Reply:
Actually think we should start at `xxxlarge` given that is the loosest size used for page stacks, and then we decide if we should open up control later only if required.